divide_and_conquer/strassen_matrix_multiplication.cpp

```diff
@@ -0,0 +1,457 @@
+#include <cassert>
```

Comment:
Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..